libs/application/templates/estate/src/fields/EstateMembersRepeater/index.tsx

```diff
@@ -19,16 +21,69 @@ import {
   SelectController,
 } from '@island.is/shared/form-fields'
 import { format as formatNationalId } from 'kennitala'
-import { EstateTypes, relationWithApplicant } from '../../lib/constants'
+import {
+  EstateTypes,
+  heirAgeValidation,
+  relationWithApplicant,
+} from '../../lib/constants'
 
 export const EstateMembersRepeater: FC<
   React.PropsWithChildren<FieldBaseProps<Answers>>
-> = ({ application, field, errors }) => {
+> = ({ application, field, errors, setBeforeSubmitCallback }) => {
   const { id } = field
   const { formatMessage } = useLocale()
+  const { getValues, setError } = useFormContext()
   const { fields, append, remove, update, replace } = useFieldArray({
     name: id,
   })
+  const values = getValues()
+
+  const selectedEstate = application.answers.selectedEstate
+
+  const hasEstateMemberUnder18 = values.estate?.estateMembers?.some(
+    (member: EstateMember) => {
+      const memberAge = kennitala.info(member.nationalId)?.age
+      return memberAge < 18 && member?.nationalId && member.enabled
+    },
+  )
+
+  const hasEstateMemberUnder18withoutRep = values.estate?.estateMembers?.some(
+    (member: EstateMember) => {
+      const advocateAge =
+        member.advocate && kennitala.info(member.advocate.nationalId)?.age
+      return (
+        hasEstateMemberUnder18 &&
+        member?.advocate?.nationalId &&
+        advocateAge &&
+        advocateAge < 18
+      )
+    },
+  )
```

Comment:
Is this enough?
Let's say there are two estate members under 18.

Then we are only finding `some` advocate where they have an advocate over 18.
This will allow an application where two members are under 18 but only one of them has an advocate over 18.

Maybe some other frontend checks will stop this from happening?

What about something like (quick pseudocode):
```typescript
const youngHeirsWithoutRep = estateMembers.filter(member => {
  if(member?.advocate && kennitala.info(member?.nationalId)?.age < 18) {
    const advocateHasNoKennitala = !member.advocate?.nationalId
    const advocateUnder18 = kennitala.info(member.advocate.nationalId).age < 18
    return advocateHasNoKennitala || advocateUnder18
  }
  return false
}
```

Then we just check whether any members remain in youngHeirsWithoutRep?

Reply:
not sure i follow, there is an alert when you enter a nationalId < 18 but it's only visual, this check is only when you try to submit, if there some member without an advocate

Reply:
if hasEstateMemberUnder18withoutRep is true theres an error at the bottom of the form
```
 if (
        hasEstateMemberUnder18withoutRep &&
        selectedEstate !== EstateTypes.divisionOfEstateByHeirs
      ) {
        setError(heirAgeValidation, {
          type: 'custom',
        })
  ```

Reply:
Ah ok, if it's just visual then.

Reply:
Ah, ok.